### ISSUE TITLE
Validate explicit ACL modifiers for top-level declarations

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#58]
Top-level declarations should use explicit ACL keywords (`public`, `internal`, `private`).

**Comment by keith** [comment]
It looks like there's no difference to sourcekit whether or not internal is specified. Using sourcekitten's structure the only difference between:

```swift
class Foo {}
```

And:

```swift
internal class Foo {}
```

Is the offset. This would have to be handled by examining the file as well.

**Comment by segiddins** [comment]
@keith examining the contents of the declaration would probably suffice for this?

**Comment by keith** [comment]
It seems like the class declaration key's offset starts after the modifiers. So in the second example from above the offset is 9. This could cause issue for cases like:

```swift
internal final class Foo {}
```

Or something crazier like:

```swift
internal
class Foo {}
```

**Comment by jpsim** [comment]
Here's a rough algorithm to do this:

1. find all top-level `internal` type declarations: see https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/TypeNameRule.swift.
2. find all `internal` modifiers: `File.matchPattern("internal", withSyntaxKinds: [.AttributeBuiltin])`.
3. throw style violations for declarations found in step 1 with no associated `internal` modifier from step 2.

**Comment by jdmarshall90** [comment]
Could this be enhanced to include *all* declarations, and not just top-level ones?
